pylint/extensions/typing.py

```diff
@@ -135,49 +122,38 @@ class TypingChecker(BaseChecker):
         ),
     )
 
+    _should_check_typing_alias: bool
+    """The use of type aliases (PEP 585) requires Python 3.9
+    or Python 3.7+ with postponed evaluation.
+    """
+
+    _should_check_alternative_union_syntax: bool
+    """The use of alternative union syntax (PEP 604) requires Python 3.10
+    or Python 3.7+ with postponed evaluation.
+    """
+
     def __init__(self, linter: PyLinter) -> None:
         """Initialize checker instance."""
         super().__init__(linter=linter)
         self._alias_name_collisions: Set[str] = set()
         self._consider_using_alias_msgs: List[DeprecatedTypingAliasMsg] = []
 
-    @lru_cache()
-    def _py37_plus(self) -> bool:
-        return self.config.py_version >= (3, 7)
-
-    @lru_cache()
-    def _py39_plus(self) -> bool:
-        return self.config.py_version >= (3, 9)
+    def open(self) -> None:
```

Comment:
I like this change, we did not need lru cache for that. It could even be faster because we're not making a function call ? Anyway, that's great.

Reply:
Don't really know if it's faster. Probably not by much I imagine.
I didn't know about `open` when I first wrote the cached functions.